src/metabase/pulse.clj

```diff
@@ -91,12 +91,22 @@
     (catch Throwable e
         (log/warn e (trs "Error running query for Card {0}" card-or-id)))))
 
+(defn- dashcard-comparator
+  "Comparator that determines which of two dashcards comes first in the layout order used for pulses.
+  This is the same order used on the frontend for the mobile layout. Orders cards left-to-right, then top-to-bottom"
+  [dashcard-1 dashcard-2]
+  (if-not (= (:row dashcard-1) (:row dashcard-2))
+    (compare (:row dashcard-1) (:row dashcard-2))
+    (compare (:col dashcard-1) (:col dashcard-2))))
+
 (defn execute-dashboard
   "Execute all the cards in a dashboard for a Pulse"
   [{pulse-creator-id :creator_id, :as pulse} dashboard-or-id & {:as options}]
   (let [dashboard-id (u/the-id dashboard-or-id)
-        dashboard (Dashboard :id dashboard-id)]
-    (for [dashcard (db/select DashboardCard :dashboard_id dashboard-id, :card_id [:not= nil])]
+        dashboard (Dashboard :id dashboard-id)
+        dashcards         (db/select DashboardCard :dashboard_id dashboard-id, :card_id [:not= nil])
+        ordered-dashcards (sort dashcard-comparator dashcards)]
```

Comment:
nitpicky but it's a little weird to align some of the `let` args but not others. Either align them all or align none of them